test/e2e/e2e_docker_helper.go

```diff
@@ -213,6 +231,58 @@ func setupByoDockerHost(ctx context.Context, clusterConName, byoHostName, namesp
 	return output, byohost.ID, err
 }
 
+func SetupByoDockerHostWithConfig(ctx context.Context, byoHostName, port, namespace string, dockerClient *client.Client, kubeconfigFile *os.File, flags map[string]string) (types.HijackedResponse, string, error) {
```

Comment:
This and SetupByoDockerHost is almost duplicate code. Can we try and make it as a single func with different configurations? 
The function arguments are exploding, so good time to define a struct?

On a side note - this file need not be called e2e_docker_helper anymore. Maybe just docker_helper will do.